src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,39 @@
-'use strict';
-
 class Animal {
-  // write your code here
+  static alive = [];
+  health = 100;
+
+  constructor(name) {
+    this.name = name;
+    Animal.alive.push(this);
+  }
+
+  removeAnimal(animal) {
+    const animalIndex = Animal.alive.indexOf(animal);
+
+    Animal.alive.splice(animalIndex, 1);
+  }
 }
 
 class Herbivore extends Animal {
-  // write your code here
+  hidden = false;
+
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  bite(herbivore) {
+    if (herbivore instanceof Carnivore || herbivore.hidden) {
```

Comment:
It's enough to check that incoming object inside `bite` method is an instance of class `Herbivore` and is not hidden within a single `if` statement. There is no need to check if it is an instance of `Carnivore` as the method should only work on `Herbivore` instances.